### PR TITLE
add environmental DNS code and variables

### PR DESCRIPTION
**Pull request by nickdevdfe**

### CODE REVIEW

File: terraform/custom_domains/environmental_domains/terraform.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+terraform {
+
+  required_version = "= 1.3.5"
```
Comment (thomasleese) [inline terraform/custom_domains/environmental_domains/terraform.tf:3]: Could we use the same version of Terraform we use elsewhere?

File: terraform/custom_domains/environmental_domains/variables.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+variable "front_door_name" {}
+variable "resource_group_name" {}
+variable "domains" {}
+variable "environment_short" {}
+variable "app_name" {}
+variable "app_resource_group_name" {}
+variable "app_sub_id" {}
+variable "redirect_rules" {
+  default = {}
+}
```
Comment (thomasleese) [inline terraform/custom_domains/environmental_domains/variables.tf:11]: Could we add a `type` and `description` to these variables?

File: terraform/custom_domains/environmental_domains/workspace_variables/afqts_test_backend.tfvars [hunk 3]
```diff
@@ -0,0 +1,4 @@
+resource_group_name  = "s189p01-afqtsdomains-rg"
+storage_account_name = "s189p01afqtsdomainstf"
```
Comment (thomasleese) [inline terraform/custom_domains/environmental_domains/workspace_variables/afqts_test_backend.tfvars:2]:
```suggestion
storage_account_name = "s189p01afqtsdomainstfsa"
```

I believe the convention is to suffix storage account names with `sa`.

**Review by thomasleese — COMMENTED** [review]
I don't think we need the `afqts_` prefix on the workspaces, since we're within the Apply for QTS project, and we don't often use the string `afqts` unless we really have to due to name length constraints.

File: terraform/custom_domains/environmental_domains/workspace_variables/afqts_test_backend.tfvars (hunk 3 above)
Comment (nickdevdfe) [reply terraform/custom_domains/environmental_domains/workspace_variables/afqts_test_backend.tfvars]: This storage account has already been deployed, and would require a new ticket to change.

File: Makefile [hunk 4]
```diff
@@ -55,8 +55,10 @@ production: paas ## Specify production PaaS environment
 	$(eval AZURE_BACKUP_STORAGE_ACCOUNT_NAME=s165p01afqtsdbbackuppd)
 	$(eval AZURE_BACKUP_STORAGE_CONTAINER_NAME=apply-for-qts)
 
+
+
 .PHONY: development_aks
-development_aks: aks ## Specify dev AKS environment
+development_aks: aks ## Specify production aks environment
```
Comment (smneal) [inline Makefile:61]: dev AKS

File: terraform/custom_domains/environment_domains/main.tf [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+data "azurerm_cdn_frontdoor_profile" "main" {
+  name                = var.front_door_name
+  resource_group_name = var.resource_group_name
+}
+
+# data "azurerm_linux_web_app" "app" {   # REMOVED
+#   provider            = azurerm.app_subcription
+#   name                = var.app_name
+#   resource_group_name = var.app_resource_group_name
+# }
```
Comment (smneal) [inline terraform/custom_domains/environment_domains/main.tf:23]: Delete?

File: terraform/custom_domains/environment_domains/variables.tf [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    type = string
+    description = "Environment"
+}  # added from tra
+variable "environment_short" {
+    type = string
+    description = "Short name for environment"
+}
+# variable "app_name" {}
+# variable "app_resource_group_name" {}
+# variable "app_sub_id" {}
```
Comment (smneal) [inline terraform/custom_domains/environment_domains/variables.tf:27]: Delete?

File: terraform/custom_domains/environmental_domains/variables.tf (hunk 2 above)
Comment (nickdevdfe) [reply terraform/custom_domains/environmental_domains/variables.tf]: Updated with type and description.

File: terraform/custom_domains/environment_domains/main.tf (hunk 5 above)
Comment (nickdevdfe) [reply terraform/custom_domains/environment_domains/main.tf:23]: Deleted.

File: terraform/custom_domains/environment_domains/terraform.tf [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+  skip_provider_registration = true
+}
+
+provider "azurerm" {
+  features {}
+  version         = "3.45.0"
+  alias           = "app_subcription"
+  subscription_id = var.app_sub_id # not in tra
+}
```
Comment (thomasleese) [inline terraform/custom_domains/environment_domains/terraform.tf:25]: It looks like we don't use this provider anywhere - can we remove it and the dependent variables?

File: .prettierignore [hunk 8]
```diff
@@ -31,3 +31,7 @@ public/assets
 storage
 tmp
 vendor
+terraform/custom_domains/environment_domains/workspace_variables/afqts_dev.tfvars.json
+terraform/custom_domains/environment_domains/workspace_variables/afqts_preprod.tfvars.json
+terraform/custom_domains/environment_domains/workspace_variables/afqts_production.tfvars.json
+terraform/custom_domains/environment_domains/workspace_variables/afqts_test.tfvars.json
```
Comment (thomasleese) [inline .prettierignore:37]: We shouldn't be ignoring these files, can we ensure they pass the linter instead?

File: terraform/custom_domains/environment_domains/main.tf [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  }
+
+  conditions {
+    host_name_condition {
+      operator     = "Equal"
+      match_values = [each.key]
+    }
+  }
+}
```
Comment (thomasleese) [inline terraform/custom_domains/environment_domains/main.tf:53]: It looks like we're not actually using this anywhere, `var.redirect_rules` is always set to `{}` - can we remove it? We can always add it back later if we learn that we need it.

File: terraform/custom_domains/environment_domains/terraform.tf [hunk 10]
```diff
@@ -0,0 +1,25 @@
+terraform {
+
+  required_version = "= 1.4.5"
```
Comment (thomasleese) [inline terraform/custom_domains/environment_domains/terraform.tf:3]: This should be `1.5.0` like the rest of the Terraform in this repo.

File: terraform/custom_domains/environment_domains/variables.tf [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  type        = string
+  description = "Environment"
+} # added from tra
+variable "environment_short" {
+  type        = string
+  description = "Short name for environment"
+}
+# variable "app_name" {}
+# variable "app_resource_group_name" {}
+# variable "app_sub_id" {}
```
Comment (thomasleese) [inline terraform/custom_domains/environment_domains/variables.tf:27]: Can we remove these?

File: terraform/custom_domains/environment_domains/terraform.tf (hunk 7 above)
Comment (nickdevdfe) [reply terraform/custom_domains/environment_domains/terraform.tf]: This has been removed

File: .prettierignore (hunk 8 above)
Comment (nickdevdfe) [reply .prettierignore]: Files have been successfully linted against Prettier and updated.

File: terraform/custom_domains/environment_domains/main.tf (hunk 9 above)
Comment (nickdevdfe) [reply terraform/custom_domains/environment_domains/main.tf]: Removed unneeded resource block and associated variables.

File: terraform/custom_domains/environment_domains/variables.tf [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  type        = string
+  description = "Name of resouce group name"
+}
+variable "domains" {
+  description = "List of domains record names"
+}
+variable "environment_tag" {
+  type        = string
+  description = "Environment"
+} # added from tra
```
Comment (thomasleese) [inline terraform/custom_domains/environment_domains/variables.tf:19]: I'm not quite sure what this comment means here?

File: terraform/custom_domains/environment_domains/workspace_variables/afqts_dev.tfvars.json [hunk 13]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (thomasleese) [inline terraform/custom_domains/environment_domains/workspace_variables/afqts_dev.tfvars.json:1]: Can we make the file names in workspace variables directory match the naming convention in the AKS workspace variables directory?

We don't need the `afqts_` prefix as we're already in this project, and we've gone with full names like `development` and `preproduction`.

File: terraform/custom_domains/environment_domains/main.tf [hunk 14]
```diff
@@ -0,0 +1,19 @@
+module "domains" {
+  source              = "git::https://github.com/DFE-Digital/terraform-modules.git//domains/environment_domains?ref=0.5.4"
```
Comment (saliceti) [inline terraform/custom_domains/environment_domains/main.tf:2]: Please use ref=stable

File: terraform/custom_domains/environment_domains/variables.tf (hunk 12 above)
Comment (nickdevdfe) [reply terraform/custom_domains/environment_domains/variables.tf:19]: removed

File: terraform/custom_domains/environment_domains/main.tf (hunk 14 above)
Comment (nickdevdfe) [reply terraform/custom_domains/environment_domains/main.tf:2]: Updated

File: terraform/custom_domains/environment_domains/workspace_variables/afqts_dev.tfvars.json (hunk 13 above)
Comment (nickdevdfe) [reply terraform/custom_domains/environment_domains/workspace_variables/afqts_dev.tfvars.json:1]: File names updated and makefile code updated to reflect filename changes.

**Review by saliceti — APPROVED** [review]

**Pull request merged**